docs/section5/Rule5-1-2.md

```diff
@@ -0,0 +1,35 @@
+
+# Envelope - Rule 5-1-2  
+
+**Rule ID:** 5-1-2  
+**Rule Description:** The building shall be modeled so that it does not shade itself.  
+**Rule Assertion:** Baseline RMR Building: surface.does_cast_shade = expected value  
+**Appendix G Section:** Section G3.1-5(b) Building Envelope Modeling Requirements for the Baseline building  
+**Appendix G Section Reference:** Table G3.1 Section 5a  
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:**  None  
+
+**Manual Check:** None  
+**Evaluation Context:** Each Data Element  
+**Data Lookup:** None  
+
+## Rule Logic:  
+
+- For each building segment in the Baseline model: ```for building_segment_b in B_RMR.building.building_segments:```  
+
+  - For each thermal_block in building segment: ```thermal_block_b in building_segment_b.thermal_blocks:```  
+  
+    - For each zone in thermal block: ```zone_b in thermal_block_b.zones:```  
+
+      - For each space in zone: ```space_b in zone_b.spaces:```  
+
+        - For each surface in space: ```for surface_b in space_b.surfaces:```  
+
+          - Check that surface is exterior: ```if ( surface_b.adjacent_to == "EXTERIOR" ): exterior_vertical_surface_b = surface_b```  
+
+          **Rule Assertion:** Baseline exterior surface does not cast shade:  
+
+          - Case 1: ```exterior_vertical_surface_b.does_cast_shade == FALSE: PASS```  
+
```

Comment:
@JasonGlazer and @mkarpman - i see that E+ has a boolean input for whether a surface is self-shading- which could be used to populate this input. However, do you know if other tools (like eQuest ) have a flag for this as well?

Reply:
There is one in eQUEST (shading-surface = yes/no)